tutorial/westeros/westeros.ipynb

```diff
@@ -0,0 +1,1268 @@
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Technological Diffusion and Contraction\n",
+    "\n",
+    "We know from historical precedent that energy systems can not be transformed instantaneously. Therefore, we use a family of constraints on **activity** (`ACT`) and **capacity** (`CAP`). "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "$\\sum_{y^V \\leq y,m} ACT_{n,t,y^V,y,m,h}\n",
+    "    \\leq ~ initial\\_activity\\_up_{n,t,y,h}\n",
+    "        \\cdot \\frac{ \\Big( 1 + growth\\_activity\\_up_{n,t,y,h} \\Big)^{|y|} - 1 }\n",
+    "                   { growth\\_activity\\_up_{n,t,y,h} }$  \n",
+    "$         + \\bigg( \\sum_{y^V \\leq y-1,m} ACT_{n,t,y^V,y-1,m,h}\n",
+    "                    + \\sum_{m} historical\\_activity_{n,t,y-1,m,h} \\bigg)$  \n",
+    "$         \\hspace{2 cm} \\cdot \\Big( 1 + growth\\_activity\\_up_{n,t,y,h} \\Big)^{|y|}$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "This example limits the ability for technologies to **grow**. It also allows new technologies to penetrate the market based on the `initial_activity_up` paramter. For the purposes of this simple model, we simply need to provide `growth_activity_up` values, which limit the rate technologies can grow."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "skip"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "## TODO: hide this\n",
+    "base_growth = {\n",
+    "    'node_loc': country,\n",
+    "    'year_act': horizon[1:],\n",
+    "    'time': 'year',\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "growth_technologies = [\n",
+    "    \"coal_ppl\", \n",
+    "    \"wind_ppl\", \n",
+    "    \"bulb\",\n",
+    "]\n",
+    "\n",
+    "for tec in growth_technologies:\n",
+    "    df = make_df(base_growth, technology=tec, value=0.05, unit='%') # TODO: is this actually right? should be fraction?\n",
+    "    scenario.add_par('growth_activity_up', df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Defining an Energy Mix"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "The simplest way to model a pre-defined energy mix is to use equality constraints on activity or capacity.\n",
+    "\n",
+    "More elegant ways include using parameters to define `historical` values, but in this simple example, we choose a starting point for the Westerosi energy system.\n",
```

Comment:
I'd go for the elegant way right away - it's one additional step, but it makes the results a lot more intuitive because it removes the big spike of new capacity in the first year

Reply:
will do!